src/jit/lclvars.cpp

```diff
@@ -1769,6 +1789,7 @@ void Compiler::lvaPromoteLongVars()
         {
             if (lvaGetPromotionType(varDsc->lvParentLcl) != PROMOTION_TYPE_INDEPENDENT)
             {
+                JITDUMP("Not promoting long local V%02u: field of dependently-promoted struct\n", lclNum);
```

Comment:
In other words, I don't think that we can consider #6778 fixed unless we have code that compensates for this limitation: liveness is going to have trouble any time it encounters the store decomposition described in that issue. It seems like that situation is still possible as long as there are long-typed non-multi-reg-{arg,return} locals that we cannot promote.